src/core/meta.ml

```diff
@@ -10,20 +10,43 @@ let rec remove m = function
 	| (m2,_,_) :: l when m = m2 -> l
 	| x :: l -> x :: remove m l
 
-let to_string m = fst (get_info m)
+type user_meta = {
+	doc : string;
+	flags : meta_parameter list;
+	source : string option;
+}
+
+let user_meta : (string, user_meta) Hashtbl.t = Hashtbl.create 0
```

Comment:
Do we clear this anywhere? Otherwise we could construct cases where we add some meta, remove it and still see it.

Reply:
I added a reset before init macros, though I'm not sure it's really clean that way..
Btw, should this data live in some kind of context instead?

Reply:
I guess it would be better if this lived on the common context (in common.ml).